real_preprocessing/src/camera_pose.cpp

```diff
@@ -0,0 +1,277 @@
+    YAML::Node tags_pix = YAML::LoadFile("detections_" + std::to_string(filenum) + ".yaml");
+    Eigen::MatrixXd w_T_cam(4, 4);
+    if(file_type==KNOWN_TAG)
+    {
+      std::vector<int>::iterator it = std::find(w_T_tags_id.begin(), w_T_tags_id.end(), tags_pix["detections"][loc]["targetID"].as<int>());
+      int index = std::distance(w_T_tags_id.begin(), it);
+      w_T_cam = w_T_tags_trans[index] * objTcam(loc, filenum);
+    }
+    if(file_type==WORLD_PRES)
+    {
+      w_T_cam=objTcam(loc,filenum);
+    }
+    worldAppend(filenum,w_T_cam);
+    for (int i = 0; i < tags_pix["detections"].size(); ++i)
+    {
+      if ((i != loc) &&
+          ((std::find(w_T_tags_id.begin(), w_T_tags_id.end(), tags_pix["detections"][i]["targetID"].as<int>()) !=
+            w_T_tags_id.end()) == false))
+      {
+        Eigen::MatrixXd obj_T_cam(4, 4);
+        obj_T_cam = objTcam(i, filenum);
+        w_T_tags_trans.push_back((w_T_cam * obj_T_cam.inverse().eval()));
+        w_T_tags_id.push_back(tags_pix["detections"][i]["targetID"].as<int>());
+        w_T_tags_size.push_back(tags_pix["detections"][i]["size"][0].as<double>());
+      }
+    }
+  }
+
+  //Returns status of file : world tag present, known tag present, all unknown tags, no file
+  int fileReader(int filenum, int& known_tag_in_file)
+  {
+    bool known_tag(false);
+    std::ifstream fin;
+    fin.open("detections_" + std::to_string(filenum) + ".yaml");
+    if (fin.is_open())
+    {
+      fin.close();
+      YAML::Node tags_pix = YAML::LoadFile("detections_" + std::to_string(filenum) + ".yaml");
+      if (filenum == 0)//polling first file?
+      {
+        known_tag_in_file = 0;
+        return WORLD_PRES; //world tag present
+      }
+      for (int i = 0; i < tags_pix["detections"].size(); ++i)
+      {
+        if (std::find(w_T_tags_id.begin(), w_T_tags_id.end(), tags_pix["detections"][i]["targetID"].as<int>()) !=w_T_tags_id.end())
+        {
+          known_tag = true;
+          known_tag_in_file = i;
+        }
+        if (tags_pix["detections"][i]["targetID"].as<int>() == world_tag) 
+        {
+          known_tag_in_file = i;
+          return WORLD_PRES; //world tag present
+        }
+      }
+      if (known_tag == true)
+      {
+        return KNOWN_TAG; //known tag present
+      }
+      else
+        return UNKNOWN; //all unknown tags
+    }
+    else 
+      return NO_FILE; //file does not exist - end of YAML list
+  }
+
+  //Polls for unknown files to be reprocessed 
+  void unknownFilepoll()
+  {
+    if (unknown_file.size() != 0) 
+    {
+      for (int i = (unknown_file.size() - 1); i >= 0; i--)
+      {
+        int known_tag_in_file;
+        if (fileReader(unknown_file[i],known_tag_in_file) == KNOWN_TAG)
+        {
+          tagCalc(unknown_file[i],known_tag_in_file, KNOWN_TAG);
+          unknown_file.erase(unknown_file.begin() + i);
+        }
+      }
+    }
+  }  
+
+  //streams all snapshot detection files for tag/pose processing 
+  void fileStream()
```

Comment:
Explore this code instead for this function:
int known_tag_in_file;
int status = fileReader(file,known_tag_in_file);
if (status != NO_FILE)
{
 if ((status == WORLD_PRES)||(status == KNOWN_TAG)) //world tag or known tags present
 {
  tagCalc(file,known_tag_in_file,status);
  unknownFilepoll();
 }
 if (status == UNKNOWN) //all unknown tags
 {
  unknown_file.push_back(file); //stores file with all unknown tags
 }
 file++
}
targetDump(w_T_tags_id,w_T_tags_size,w_T_tags_trans);